scripts/base/protocols/modbus/main.zeek

```diff
@@ -48,14 +52,28 @@ event modbus_message(c: connection, headers: ModbusHeaders, is_orig: bool) &prio
 		}
 
 	c$modbus$ts   = network_time();
+	c$modbus$tid = headers$tid;
+	c$modbus$unit = headers$uid;
 	c$modbus$func = function_codes[headers$function_code];
+
+	## If this packet is a response message, swap orig/resp because the actual modbus message
+	## doesn't follow the direction of the TCP session
+	if ( ! is_orig )
+		{
+				const orig_c = copy(c);
+
+				c$id$orig_h = orig_c$id$resp_h;
+				c$id$orig_p = orig_c$id$resp_p;
+
+				c$id$resp_h = orig_c$id$orig_h;
+				c$id$resp_p = orig_c$id$orig_p;
```

Comment:
I don't feel great about this. I am not entirely sure - but I think this might cause issues, since this means that the underlying internal connection record, and the script-level connection are now out of sync.

Perhaps we should just expose `FlipRoles` to scriptland, so that a script can set it? Or just mark the fact that the order was reversed in some other way in the modbus log.

Reply:
Re-did this - instead of swapping conn record values, added a field to indicate the PDU direction